packages/material-ui/src/Portal/Portal.js

```diff
@@ -28,13 +28,21 @@ const Portal = React.forwardRef(function Portal(props, ref) {
     }
   }, [container, disablePortal]);
 
-  React.useImperativeHandle(ref, () => mountNode || childRef.current, [mountNode]);
+  useEnhancedEffect(() => {
+    if (mountNode || disablePortal) {
+      setRef(ref, mountNode || childRef.current);
+    }
+
+    return () => {
+      setRef(null);
```

Comment:
It seems that `setRef(null)` doesn't do anything.
https://github.com/mui-org/material-ui/blob/a1ffac579efb078728188e89c8a3362fd684fc4e/packages/material-ui/src/utils/reactHelpers.js#L8-L14

Reply:
Thanks, my first iteration was crap, reworking it :)